src/jit/lsra.cpp

```diff
@@ -5005,28 +5111,87 @@ LinearScan::allocateBusyReg(Interval *current, RefPosition *refPosition)
         }
 
         if (nextLocation > physRegNextLocation)
+        {
             nextLocation = physRegNextLocation;
+        }
+
+        bool isBetterLocation;
 
-        bool isBetterLocation = (nextLocation > farthestLocation);
 #ifdef DEBUG
         if (doSelectNearest() && farthestRefPhysRegRecord != nullptr)
         {
-            isBetterLocation = !isBetterLocation;
+            isBetterLocation = (nextLocation <= farthestLocation);
         }
-#endif // DEBUG
+        else
+        // the below if-stmt is associated with this else
```

Comment:
I don't think this comment adds anything; it just further separates the `else` from the `if`, and this is a widely used pattern in the JIT, so it should not be confusing.